### PR TITLE
GAS-124: Write out .kube/config from multiple sources.

### PR DESCRIPTION
**Pull request by aaronplagge**
* Moved kubectl config tasks to their own file
* Added variables for use when pulling kubectl config from pmm server
* Added logic to pull kubectl config(s) from pmm-server if they are avaialble.  
* Can also specify which kubectl config's to pull from pmm server

### CODE REVIEW

**Review by crimethinking — APPROVED** [review]

File: roles/tools/tasks/write_pmm_kubectl_config.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+- name: Checking for .kube config file
```
Comment (cezmunsta) [inline roles/tools/tasks/write_pmm_kubectl_config.yaml:2]: @aaronplagge why is this here when it is already elsewhere https://github.com/cezmunsta-org/gascan-ansible/pull/11/files#diff-ae6a42d78cbb5225d5d0a1c72f7a78fec4178fa0a1591becbf424588651f4ff1R17-R22 ?

File: roles/tools/tasks/write_pmm_kubectl_config.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    validate_certs: '{{ pmm_server_insecure }}'
+  register: tools_kubectl_from_pmm
+
+- name: Set tools_kubectl_config
+  ansible.builtin.set_fact:
+    tools_kubectl_config: "{{ tools_kubectl_from_pmm.json.kube_auth.kubeconfig }}"
+
+- name: Write kube config from pmm to file
+  ansible.builtin.copy:
+    content: "{{ tools_kubectl_config }}"
```
Comment (cezmunsta) [inline roles/tools/tasks/write_pmm_kubectl_config.yaml:38]: @aaronplagge is this appending, or overwriting in the loop?

File: roles/tools/tasks/write_pmm_kubectl_config.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    tools_kubectl_env_config_file: "{{ tools_kubectl_config_file }}"
+
+- name: Set tools_kube_config_name
+  ansible.builtin.set_fact:
+    tools_kubectl_env_config_file: "{{ tools_kubectl_config_file }}_{{ item }}"
+  when: tools_kubectl_config_check.stat.exists
+
+- name: Retrieve kube config from pmm
+  ansible.builtin.uri:
+    url: "https://{{ pmm_server_host }}:{{ pmm_server_port }}/v1/management/DBaaS/Kubernetes/Get"
```
Comment (cezmunsta) [inline roles/tools/tasks/write_pmm_kubectl_config.yaml:18]: @aaronplagge please use the predefined path fragments, in this case it would be `pmm_server_uri` along with either `pmm_api_management` or a new one that extends to include `Kubernetes`

File: roles/tools/tasks/write_pmm_kubectl_config.yaml (hunk 1 above)
Comment (aaronplagge) [reply roles/tools/tasks/write_pmm_kubectl_config.yaml:2]: This is here as if the .kube/config file was written out previously it will change then use a name of .kube/config_<PMM_ENV_NAME> to write it out to file therefore not overwriting the existing .kube/config file.

File: roles/tools/tasks/write_pmm_kubectl_config.yaml (hunk 2 above)
Comment (aaronplagge) [reply roles/tools/tasks/write_pmm_kubectl_config.yaml:38]: Neither.  If the .kube/config file does not exist it will use that name.  Otherwise it will use the name .kube/config_<PMM_ENV_NAME> where PMM_ENV_NAME is the name given tot he kubernetes cluster in PMM.

File: roles/tools/tasks/write_pmm_kubectl_config.yaml (hunk 1 above)
Comment (cezmunsta) [reply roles/tools/tasks/write_pmm_kubectl_config.yaml:2]: https://github.com/aaronplagge/gascan-ansible/blob/GAS-124/roles/tools/tasks/kubectl.yaml#L17-L22
https://github.com/aaronplagge/gascan-ansible/blob/GAS-124/roles/tools/tasks/write_pmm_kubectl_config.yaml#L2-L5

File: roles/tools/tasks/write_pmm_kubectl_config.yaml (hunk 2 above)
Comment (cezmunsta) [reply roles/tools/tasks/write_pmm_kubectl_config.yaml:38]: https://github.com/aaronplagge/gascan-ansible/blob/GAS-124/roles/tools/tasks/write_pmm_kubectl_config.yaml#L7-L14
That sets the same fact twice

File: roles/tools/tasks/write_pmm_kubectl_config.yaml (hunk 3 above)
Comment (cezmunsta) [reply roles/tools/tasks/write_pmm_kubectl_config.yaml:18]: @aaronplagge any updates?

File: roles/tools/tasks/write_pmm_kubectl_config.yaml (hunk 2 above)
Comment (cezmunsta) [reply roles/tools/tasks/write_pmm_kubectl_config.yaml:38]: @aaronplagge any updates?

File: roles/tools/tasks/write_pmm_kubectl_config.yaml (hunk 1 above)
Comment (cezmunsta) [reply roles/tools/tasks/write_pmm_kubectl_config.yaml:2]: @aaronplagge any updates?

**Comment by cezmunsta** [comment]
Closing this off - recreate in GAS
